modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/test/js/components/LayoutProvider/handlers/fieldDeletedHandler.es.js

```diff
@@ -15,69 +15,30 @@
 import {FormSupport} from 'dynamic-data-mapping-form-renderer';
 
 import * as fieldDeletedHandler from '../../../../../src/main/resources/META-INF/resources/js/components/LayoutProvider/handlers/fieldDeletedHandler.es';
-import RulesSupport from '../../../../../src/main/resources/META-INF/resources/js/components/RuleBuilder/RulesSupport.es';
 import mockPages from '../../../__mock__/mockPages.es';
 
 describe('LayoutProvider/handlers/fieldDeletedHandler', () => {
 	describe('handleFieldDeleted(state, event)', () => {
-		it('calls removeRow() when row is left with no fields after delete operation', () => {
+		it('calls removeEmptyRows() when row is left with no fields after delete operation', () => {
 			const event = {
-				columnIndex: 0,
-				pageIndex: 0,
-				rowIndex: 0,
+				activePage: 0,
+				fieldName: 'radio',
+				removeEmptyRows: true,
 			};
 			const state = {
 				pages: mockPages,
 				rules: [],
 			};
 
-			const removeRowSpy = jest.spyOn(FormSupport, 'removeRow');
+			const removeRowSpy = jest.spyOn(FormSupport, 'removeEmptyRows');
 
 			removeRowSpy.mockImplementation(() => []);
 
-			fieldDeletedHandler.handleFieldDeleted(state, event);
+			fieldDeletedHandler.handleFieldDeleted({}, state, event);
 
 			expect(removeRowSpy).toHaveBeenCalled();
 
 			removeRowSpy.mockRestore();
 		});
-
-		it('calls clearAllConditionFieldValues() when deleting a field used as the first operand of a condition', () => {
```

Comment:
@alinedoleron, why this test is no longer necessary? Can you please clarify?

Reply:
Hi @natocesarrego, it was a mistake. I think I may be debug wrongly. But I'm still having problem with this test.
It says it failed but actually the method is being called if I debug it.

```
LayoutProvider/handlers/fieldDeletedHandler › handleFieldDeleted(state, event) › calls clearAllConditionFieldValues() when deleting a field used as the first operand of a condition

    expect(jest.fn()).toHaveBeenCalled()

    Expected number of calls: >= 1
    Received number of calls:    0

      78 | 			fieldDeletedHandler.handleFieldDeleted({}, state, event);
      79 |
    > 80 | 			expect(clearAllConditionFieldValuesSpy).toHaveBeenCalled();
         | 			                                        ^
      81 |
      82 | 			clearAllConditionFieldValuesSpy.mockRestore();
      83 | 		});

      at Object.toHaveBeenCalled (test/js/components/LayoutProvider/handlers/fieldDeletedHandler.es.js:80:44)


  console.log src/main/resources/META-INF/resources/js/components/RuleBuilder/RulesSupport.es.js:52
    entrou!!!
```

And this 'entrou!!!' I put into the method `clearAllConditionFieldValuesSpy`. =/ I don't understand the reason it is happening.

Reply:
@alinedoleron, let's screen share after our daily meeting to talk about this pull. Thanks.